### ISSUE TITLE
Visual Problem Editor enhancements (BD-14)

### ISSUE THREAD

**Issue by e0d** [openedx/platform-roadmap#13]
### Problem Editor Settings

Building out a new visual problem editor focuses on having easy access to the five most common problem types (text input, numerical response, dropdown, multi-select, single-select). These account for 99% of all problem type use cases. 

Additionally, the new problem editor will:

- Enable course authors to build problem types without markdown

- Enables architectural vision and feature parity

- The UX design enables custom visuals and accessibility which make it easier to create a course and use tools

**Status**
As of Nov 4, pre-grooming is in progress. Target March 2023 for completion.

--------

Milestone 3.2: 

Objective - Update the CAPA problem editor in the course authoring experience to use a more interactive WYSIWYG editor. In doing so, we aim to improve the problem authoring experience for partners by improving the UI of this component.

Update capa problem editor for content libraries to use new interactive problem editor contribution used on LabXchange.org currently.

Jira epic: https://openedx.atlassian.net/browse/TNL-8366

### Initiative-level Narrative

Building on the work done to support a new content storage engine for the Open edX platform in Blockstore, this effort is a major update of our unstructured content library authoring experience, as well as the content tagging infrastructure necessary to help organize and structure content for use and reuse in courses. While today library content only powers randomized assessments, this effort seeks to expand the use of course content reference linked to content libraries, bringing a flexible new way to author content without committing to course structure in Studio. Course teams will now be able to annotate and organize their authored content through the use of taxonomies, whose tags can be used for partner and internal classifications of our catalog objects. These tags provide a way for educators to source / reference content from content libraries into their courses, enabling the powerful versioning and reuse workflows currently only possible by managing hundreds of content libraries at a time, as MIT / Harvard do today.  
|Customer Impact|Details|
|---|---|
|Reach (Customer) |All course authors across the platform will have the ability to create content libraries and taxonomies. 
|Impact + Measure (Outcome)| Improvement to educator NPS rating & platform map competitiveness assessment (currently authoring is listed as a top 10 gap area) 
|T&L Gap Impact|(FY20 Q4 list)|Course authoring is a top 10 T&L platform gap area, and these changes will introduce a set of key requested features not available on any other top 10 competitive alternatives to Open edX’s authoring experience. |

**Comment by e0d** [comment]
This PR is related: https://github.com/open-craft/frontend-lib-content-components/pull/6

**Comment by jmakowski1123** [comment]
> This PR is related: [open-craft/frontend-lib-content-components#6](https://github.com/open-craft/frontend-lib-content-components/pull/6)
> 
> OpenCraft is working on some components of this work as part of a grant. We'd like to know if there's any difference between this item and the work that they are doing currently.
> 
> @jmakowski1123

@e0d The PR from OC is part of the larger milestone outlined in this ticket. edX is currently pre-grooming the remainder of the work (the three general bullet points listed in project description above).

**Comment by farhaanbukhsh** [comment]
Hello @e0d @jmakowski1123 ,

Sorry to jump in the middle 😇 

We are working to complete the CAPA problem editor and co-ordinating work with @connorhaugh, the main PR where we are merging the work is https://github.com/edx/frontend-lib-content-components/pull/110 

The PR https://github.com/open-craft/frontend-lib-content-components/pull/6 is one of the component where worked on implementing the settings component.

The work that is done till now are:

1. Getting the basic structure of CAPA problem to work with calls between edx-platform and the editor
2. Implementing the answer widget. (Merged into the main PR)
3. Implementing the settings widget. (Internal Review)
4. Implementing the question widget. (Merged into the main PR)
5. Building OLX parser to get the data from edx-platform to the editor using React states. (Merged into the main PR)
6. Build OLX Transformer to convert the React State to a properly formatted OLX. (Merged into the main PR)

The settings widget is still under internal review and mostly be done this week and merged in the upstream PR.

There are few design decisions that need to be finalized:

1. Adding tolerance in the numeric problem
2. Should we go with supporting prompt for questions

We have not implemented the "Select Problem Type" page, but we would love to implement that as well. The work that would be remaining after settings widget gets merged to the main PR is to address the upstream comments and fix any broken tests.

We have not done any work on the [Video Editor](https://github.com/openedx/platform-roadmap/issues/107) yet, but we are open to discuss this and would be excited to implement the remaining parts of it.

Please let me know if there are any clarifications needed.

**Comment by connorhaugh** [comment]
Hi all: Just to verify Farhaan's point, TNL is poised to begin work on the remaining mini-milestones in the capa problem editor. That work is soft-gated on the merging of https://github.com/edx/frontend-lib-content-components/pull/110 .The mini-milestones/features we plan to work on:
1.  Select Problem Type Screen
2. Raw OLX Editor 
3. Tolerance
4. Expanding Complex Text Areas (maybe, not blocker for release, but would add images to answers).
5. Maintenance & Upkeep 

Note: we also are currently moving frontend-lib-content-components into the openedx org.

**Comment by sarina** [comment]
Closing as this is marked "Shipped" on the PR board 🎉
